src/v/cluster/node/types.cc

```diff
@@ -42,4 +42,60 @@ std::ostream& operator<<(std::ostream& o, const local_state& s) {
     return o;
 }
 
-} // namespace cluster::node
\ No newline at end of file
+} // namespace cluster::node
+
+namespace reflection {
+template<typename T>
+void read_and_assert_version(std::string_view type, iobuf_parser& parser) {
+    auto version = adl<int8_t>{}.from(parser);
+    vassert(
+      version <= T::current_version,
+      "unsupported version of {}, max_supported version: {}, read version: {}",
+      type,
+      version,
+      T::current_version);
+}
+
+void adl<cluster::node::disk>::to(iobuf& out, cluster::node::disk&& s) {
+    serialize(out, s.current_version, s.path, s.free, s.total);
+}
+
+cluster::node::disk adl<cluster::node::disk>::from(iobuf_parser& p) {
+    read_and_assert_version<cluster::node::disk>("cluster::node::disk", p);
+
+    auto path = adl<ss::sstring>{}.from(p);
+    auto free = adl<uint64_t>{}.from(p);
+    auto total = adl<uint64_t>{}.from(p);
+
+    return cluster::node::disk{
+      .path = path,
+      .free = free,
+      .total = total,
+    };
+}
+
+void adl<cluster::node::local_state>::to(
+  iobuf& out, cluster::node::local_state&& r) {
+    reflection::serialize(
+      out,
+      r.current_version,
+      std::move(r.redpanda_version),
+      r.uptime,
+      std::move(r.disks));
+}
+
+cluster::node::local_state
+adl<cluster::node::local_state>::from(iobuf_parser& p) {
+    read_and_assert_version<cluster::node::local_state>(
+      "cluster::node::local_state", p);
+
+    auto redpanda_version = adl<cluster::node::application_version>{}.from(p);
+    auto uptime = adl<std::chrono::milliseconds>{}.from(p);
+    auto disks = adl<std::vector<cluster::node::disk>>{}.from(p);
+
+    return cluster::node::local_state{
+      .redpanda_version = std::move(redpanda_version),
+      .uptime = uptime,
+      .disks = std::move(disks)};
+}
```

Comment:
the commit message says `cluster/health: move node-local serialization` but these two methods don't seem to be moved from anywhere. are they new or was there a corresponding deletion missing?

Reply:
looking a little closer it doesn't seem like `cluster::node::local_state` is serialized in this patch set. so presumably you could introduce these two methods in a later PR?

Reply:
Yep, good catch. Splitting up the original patch was a bit more awkward than I expected--and 
I ended up ditching the serialization version bump patch that used it.